The default config files are located in /etc/default in oe-core. Update to make
the checkbox 'Allow root SSH login with password' work.

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Kai Kang <kai.kang@windriver.com>

Rebase for anaconda 34.

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
 pyanaconda/modules/users/installation.py | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/pyanaconda/modules/users/installation.py b/pyanaconda/modules/users/installation.py
index 36fbae0b1..e389cec09 100644
--- a/pyanaconda/modules/users/installation.py
+++ b/pyanaconda/modules/users/installation.py
@@ -160,7 +160,7 @@ class SetSshKeysTask(Task):
 class ConfigureRootPasswordSSHLoginTask(Task):
     """Optionally add an override allowing root to login with password over SSH."""
 
-    CONFIG_PATH = "etc/ssh/sshd_config.d/01-permitrootlogin.conf"
+    CONFIG_PATH = "etc/default/sshd-permitrootlogin"
 
     def __init__(self, sysroot, password_allowed):
         """Create a new root password SSH login configuration task.
@@ -186,7 +186,13 @@ class ConfigureRootPasswordSSHLoginTask(Task):
                 f.write(
                     '# This file has been generated by the Anaconda Installer.\n'
                     '# Allow root to log in using ssh. Remove this file to opt-out.\n'
-                    'PermitRootLogin yes\n'
+                    'SSHD_OPTS="-oPermitRootLogin=yes"\n'
                 )
         else:
             log.debug("Not adding an override allowing root login with password via SSH.")
+            with open(os.path.join(self._sysroot, self.CONFIG_PATH), "wt") as f:
+                f.write(
+                    '# This file has been generated by the Anaconda Installer.\n'
+                    '# Disallow root to log in using ssh. Remove this file to opt-out.\n'
+                    'SSHD_OPTS="-oPermitRootLogin=no"\n'
+                )
